modules/plugin/geopkg/src/main/java/org/geotools/geopkg/GeoPackage.java

```diff
@@ -308,56 +309,66 @@ static void createFunctions(Connection cx) throws SQLException {
         Function.create(
                 cx,
                 "ST_MinX",
-                new GeometryFunction() {
-                    @Override
-                    public Object execute(GeoPkgGeomReader reader) throws IOException {
+                new GeometryDoubleFunction() {
+
+                    public double execute(GeoPkgGeomReader reader) throws IOException {
                         return reader.getEnvelope().getMinX();
                     }
-                });
+                },
```

Comment:
Not sure about the meaning of the "1" value parameter but I have noticed it's in all the other calls except this one.

Reply:
Oh right, needs to be added, it tells sqlite how many params the function will take, so that it can do the arg count check by itself.